src/UNIXSocketRequest.cpp

```diff
@@ -12,13 +12,18 @@
 #include "UNIXSocketRequest.hpp"
 #include "factoryRequestImplemetator.hpp"
 #include "urlRequest.hpp"
+#include <string>
+#include <unordered_set>
 
 using wrapperType = cURLWrapper;
 
 void UNIXSocketRequest::download(const URL& url,
                                  const std::string& outputFile,
-                                 std::function<void(const std::string&, const long)> onError)
+                                 std::function<void(const std::string&, const long)> onError,
+                                 const std::unordered_set<std::string>& httpHeaders)
 {
+    std::ignore = httpHeaders;
```

Comment:
I think it would be better to use an attribute
```suggestion
                                 [[maybe_unused]] const std::unordered_set<std::string>& httpHeaders)
{
```

Reply:
Done.